fedbiomed/researcher/federated_workflows/_experiment.py

```diff
@@ -847,36 +865,159 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
         return 1
 
     def _collect_optim_aux_var(
-            self,
-    ) -> Optional[Dict[str, Dict[str, Any]]]:
+        self,
+    ) -> Optional[Dict[str, AuxVar]]:
         """Collect auxiliary variables of the held Optimizer, if any."""
         if self._agg_optimizer is None:
             return None
         return self._agg_optimizer.get_aux()
 
-    def _process_optim_aux_var(
+    def _aggregate_cleartext_optim_auxvar(
         self,
-        aux_var: Dict[str, Dict[str, Dict[str, Any]]]
-    ) -> None:
-        """Process Optimizer auxiliary variables received during last round.
+        nodes_auxvar: Dict[str, Dict[str, AuxVar]],
+    ) -> Dict[str, AuxVar]:
+        """Aggregate clear-text node-wise optimizer auxiliary variables.
+
+        Args:
+            nodes_auxvar: Dict of node-wise optimizer auxiliary variables,
+                with format `{node_name: {module_name: module_auxvar}}`.
+
+        Returns:
+            Aggregated optimizer auxiliary variables, with format
+            `{module_name: module_auxvar}`.
 
         Raises:
-            FedbiomedExperimentError: if auxiliary variables were received,
-                but `agg_optimizer` is None and thus cannot process them.
-            FedbiomedOptimizerError: if the received auxiliary variables do
-                not match the expectations of the `agg_optimizer` (Aggregation) Optimizer.
+            FedbiomedExperimentError: If any node sent unproper-type data.
         """
-        # If an Optimizer is used, pass it the auxiliary variables (if any).
-        if self._agg_optimizer is not None:
-            self._agg_optimizer.set_aux(aux_var)
-        # If no Optimizer is used but auxiliary variables were received, raise.
-        elif aux_var:
+        failed = []  # type: List[str]
+        auxvar = {}  # type: Dict[str, AuxVar]
+        for node_name, node_dict in nodes_auxvar.items():
+            if not isinstance(node_dict, dict):
+                failed.append(node_name)
+            elif not failed:
+                for mod_name, mod_auxv in node_dict.items():
+                    auxvar[mod_name] = auxvar.get(mod_name, 0) + mod_auxv
+        if failed:
             raise FedbiomedExperimentError(
-                "Received auxiliary variables from 1+ node Optimizer, but "
-                "no `agg_optimizer` was set for this Experiment to process "
-                "them.\nThese variables come from the following plug-in "
-                f"modules: {set(aux_var)}."
+                "Received unproper-type optimizer auxiliary variables from "
+                "1+ nodes.\nNodes that sent invalid data are the following: "
+                f"{failed}."
+            )
+        return auxvar
+
+    def _aggregate_encrypted_model_params_and_optim_auxvar(
+        self,
+        model_params: Dict[str, List[int]],
+        optim_auxvar: Dict[str, EncryptedAuxVar],
```

Comment:
This method is in charge of too much things. Moreover, it should be split across `federated_workflow`, for other uses (think of federated analytics)